install_config/persistent_storage/persistent_storage_csi.adoc

```diff
@@ -0,0 +1,110 @@
+
+Following picture provides high-level overview about these components running in pods in {product-title} cluster.
+
+image::csi.png["Architecture of CSI components"]
+
+It is possible to run multiple CSI drivers for different storage backends. Each driver needs its own external attacher StatefulSet, external provisioner Deployment and DaemonSet with the driver and CSI registrar.
+
+=== External CSI attacher StatefulSet
+
+External CSI attacher is a StatefulSet that deploys a single pod with two containers:
+
+  * External CSI attacher container itself that translates `attach` and `detach` calls from {product-title} to respective `ControllerPublish` and `ControllerUnpublish` calls to CSI driver.
+  * CSI driver container.
+
+CSI attacher container talks to CSI driver container using Unix Domain Sockets, ensuring that no CSI communication leaves the pod. CSI driver is not accessible from outside of the pod at all.
+
+StatefulSet is used to ensure that exactly one instance of the pod runs at any given time. External attacher pod does not scale in {product-title} {product-version}.
+
+[NOTE]
+====
+`attach` and `detach` operations typically require CSI driver to use credentials to the storage backend. It is recommended to run external attacher pod runs xref:../../install_config/install/advanced_install.adoc#configuring-dedicated-infrastructure-nodes[infrastructure nodes] so the credentials can never leak to user processes even in case of catastrophic security breach on a node.
+====
+
+[NOTE]
+====
+External attacher must run also for CSI drivers that don't support 3rd party attach/detach operations. External attacher will not issue any `ControllerPublish` or `ControllerUnpublish` to CSI driver, however it still must run to implement necessary {product-title} attachment API.
+====
+
+=== External CSI provisioner deployment
+
+External CSI provisioner is a Deployment that deploys single or several pods with two containers:
+
+  * External CSI provisioner container itself that translates `provision` and `delete` calls from {product-title} to respective `CreateVolume` and `DeleteVolume` calls to CSI driver.
+  * CSI driver container.
+
+CSI provisioner container talks to CSI driver container using Unix Domain Sockets, ensuring that no CSI communication leaves the pod. CSI driver is not accessible from outside of the pod at all.
+
+Deployment can be used to scale up/down number of provisioners.
+
+[NOTE]
+====
+`provision` and `delete` operations typically require CSI driver to use credentials to the storage backend. It is recommended to run external provisioner pods on xref:../../install_config/install/advanced_install.adoc#configuring-dedicated-infrastructure-nodes[infrastructure nodes] so the credentials can never leak to user processes even in case of catastrophic security breach on a node.
```

Comment:
s/require CSI driver/require the CSI driver

The second sentence is not clear. Should it be "It is recommended to run the external attacher pod infrastructure nodes so that the credentials never leak to the user processes, even in the event of a catastrophic security breach on a node."